py3status/modules/arch_updates.py

```diff
@@ -82,8 +84,10 @@ def check_updates(self):
         if self.include_pacman:
             pacman_updates = self._check_pacman_updates()
 
-        if self.include_aur:
-            aur_updates = self._check_aur_updates()
+        if self.include_aur and self.py3.check_commands(['cower']):
+            aur_updates = self._check_aur_updates_cower()
+        elif self.include_aur and self.py3.check_commands(['yay']):
+            aur_updates = self._check_aur_updates_yay()
```

Comment:
this work (deciding which update method to use) should be done in `post_config_hook()` because then we only need to do it once.

the simplest way would be something like

post_config_hook()
```python
if self.include_aur:
  if self.py3.check_commands(['cower']):
    self._check_aur_updates = self._check_aur_updates_cower
  elif self.py3.check_commands(['yay']):
    self._check_aur_updates = self._check_aur_updates_yay
  else:
    self.py3.notify_user('cower/yay is not installed cannot check aur')
```

check_updates()
```python
  if self.include_aur:
    aur_updates = self._check_aur_updates()
```

Reply:
Thanks for the review and your suggestion. I updated my commit.